core/state-machine/src/lib.rs

```diff
@@ -123,8 +176,8 @@ pub trait Externalities<H: Hasher> {
 	}
 
 	/// Clear a child storage entry (`key`) of current contract being called (effective immediately).
-	fn clear_child_storage(&mut self, storage_key: &[u8], key: &[u8]) -> bool {
-		self.place_child_storage(storage_key.to_vec(), key.to_vec(), None)
+	fn clear_child_storage(&mut self, storage_key: ChildStorageKey<H>, key: &[u8]) {
```

Comment:
not related to this PR, but it is somewhat weird to make the argument `key: &[u8]` if it is just to convert to vec then.
you probably shouldn't change it here though.